x-pack/plugins/reporting/server/plugin.test.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+jest.mock('./browsers/install', () => ({
+  installBrowser: jest.fn().mockImplementation(() => ({
+    binaryPath$: {
+      pipe: jest.fn().mockImplementation(() => ({
+        toPromise: () => Promise.resolve(),
+      })),
+    },
+  })),
+}));
+
+import { coreMock } from 'src/core/server/mocks';
+import { ReportingPlugin } from './plugin';
+import { createMockConfig } from './test_helpers';
+
+const sleep = (time: number) => new Promise((r) => setTimeout(r, time));
+
+describe('Reporting Plugin', () => {
+  let config: any;
+  let initContext: any;
+  let coreSetup: any;
+  let coreStart: any;
+  let pluginSetup: any;
+  let pluginStart: any;
+
+  beforeEach(async () => {
+    config = createMockConfig();
+    initContext = coreMock.createPluginInitializerContext(config);
+    coreSetup = await coreMock.createSetup(config);
+    coreStart = await coreMock.createStart();
+    pluginSetup = ({
+      licensing: {},
+      usageCollection: {
+        makeUsageCollector: jest.fn(),
+        registerCollector: jest.fn(),
+      },
+      security: {
+        authc: {
+          getCurrentUser: () => ({
+            id: '123',
+            roles: ['superuser'],
+            username: 'Tom Riddle',
+          }),
+        },
+      },
+    } as unknown) as any;
+    pluginStart = ({
+      data: {
+        fieldFormats: {},
+      },
+    } as unknown) as any;
+  });
+
+  it('has a sync setup process', () => {
+    const plugin = new ReportingPlugin(initContext);
+
+    expect(plugin.setup(coreSetup, pluginSetup)).not.toHaveProperty('then');
+  });
+
+  it('logs setup issues', async () => {
+    const plugin = new ReportingPlugin(initContext);
+    // @ts-ignore overloading error logger
+    plugin.logger.error = jest.fn();
+    coreSetup.elasticsearch = null;
+    plugin.setup(coreSetup, pluginSetup);
+
+    await sleep(5);
+
+    // @ts-ignore overloading error logger
+    expect(plugin.logger.error.mock.calls).toMatchInlineSnapshot(`
```

Comment:
how about:
```
expect(plugin.logger.error.mock.calls).toMatch(/Error in Reporting setup, reporting may not function properly/)
```

I believe `toMatch` can work on objects, and it basically lets you regex match against a stringified version of the object.